### PR TITLE
chore: revert attrs dependency changes

### PR DESCRIPTION
**Pull request by sauyon**
We only need attrs 21.1 for type checking.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -81,6 +81,7 @@ docs = ["recommonmark", "sphinx", "sphinx-click", "sphinx_rtd_theme", "sphinxcon
 types = ["types-click", "types-chardet", "types-setuptools", "pyspark-stubs", "types-filelock", "types-PyYAML"]
 
 [tool.poetry.dev-dependencies]
+attrs = ">=21.1"
```
Comment (aarnphm) [inline pyproject.toml:84]: but then I think by default this will get install nonetheless, so it will override with line 50

Comment (sauyon) [reply pyproject.toml:84]: Yes, but it won't be an issue if user packages require old attrs for some reason.

**Comment by sauyon** [comment]
I've moved the type packages into poetry's dev dependencies; I'm pretty sure those aren't included in pypi packages published by poetry.

**Review by bojiang — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (bojiang) [reply pyproject.toml:77]: (really detailed)

**Pull request merged**